notebook/notebookapp.py

```diff
@@ -576,7 +584,7 @@ class NotebookApp(JupyterApp):
     flags = flags
     
     classes = [
-        KernelManager, Session, MappingKernelManager, KernelSpecManager,
+        KernelManager, Session, MappingKernelManager, AsyncMappingKernelManager, KernelSpecManager,
```

Comment:
Are all traits between `MappingKernelManager` and `AsynMappingKernelManager` the same? It seems confusing to always expose both sets of traits to the CLI if you can only use one or the other.

Reply:
With the exception of the default value for [`kernel_manager_class`](https://github.com/kevin-bates/notebook/blob/async-subclasses/notebook/services/kernels/kernelmanager.py#L266) - yes - they are the same.  All traits are physically defined on the `MappingKernelManagerBase` mixin.

My concern was not to impact existing configs - which will be using `MappingKernelManager` as the "trait class".

I'm sure there's some trailet voodoo that we can take advantage of but I'm not up on how to go about doing this.  How can a trait associated with `MappingKernelManager` (albeit defined on the mixin) be available to `AsyncMappingKernelManager`?

Reply:
I'm diving into the code now to explore this further. We need `MappingKernelManager` to be inserted somewhere in `AsyncMappingKernelManager`'s MRO, but we need to do this cautiously. I know you've explored this already, so I likely will reach your same conclusion haha. But maybe it's worth having someone else look at the code more deeply.

Reply:
ok - thanks.  That doesn't feel like the right thing to do from a class hierarchy perspective, but let's see where it takes this.